src/ConnectedMode/QualityProfiles/QualityProfileUpdater.cs

```diff
@@ -37,10 +39,27 @@ internal interface IQualityProfileUpdater
     [PartCreationPolicy(CreationPolicy.Shared)]
     internal class QualityProfileUpdater : IQualityProfileUpdater
     {
-        public Task UpdateAsync()
+        private readonly IOutOfDateQPFinder outOfDateQPFinder;
+        private readonly ILogger logger;
+
+        [ImportingConstructor]
+        public QualityProfileUpdater(IOutOfDateQPFinder outOfDateQPFinder,
+            ILogger logger)
+        {
+            this.outOfDateQPFinder = outOfDateQPFinder;
+            this.logger = logger;
+        }
+
+        public async Task UpdateAsync()
         {
-            // TODO - implement
-            return Task.CompletedTask;
+            var languagesToUpdate = await outOfDateQPFinder.GetAsync(CancellationToken.None);
+
+            if (!languagesToUpdate.Any())
+            {
+                logger.LogQPVerbose("No updates required");
+            }
+
+            // TODO - update and save changed profiles
```

Comment:
The code that does the "fetch and save" during binding is in [BindingProcessImpl](https://github.com/SonarSource/sonarlint-visualstudio/blob/154183e913b6a4f2740b601e171835035f20b641/src/ConnectedMode/Binding/BindingProcessImpl.cs#L101). I'm looking at how to refactor/reuse that code so it can be used both for a new binding and for updating QPs.